alertbox/devcontainers-try-dotnet
# fix: dev container not working due to user permission issue

this changeset is to fix the dev container due to a user permission issue causing it to spin up.

- add powershell feature
- add dotnet latest feature
- remove common utilities for mixing container users (https://github.com/devcontainers/features/issues/440)
- remove azure functions core tools for not supporting arm64 (https://github.com/Azure/azure-functions-core-tools/issues/3112)
- add env variables for functions
- add ai-based intellisense vs code extension
- add vs code extensions for azure
- add vs code extensions for infra as a code
- add ms sql vs code extension
- rearrange vs code extensions
- add vs code settings for azure functions
- add comments about verifying the contents of the box
- up-to-date repo doc according to dev container contents

BREAKING CHANGE: this version does not support azure functions on arm64 due to the unavailability of binaries.
